src/query.rs

```diff
@@ -550,26 +668,98 @@ mod tests {
             );
         }
 
+        let transparent_actual_results: BTreeMap<_, Vec<BTreeMap<_, TransparentValue>>> =
+            actual_results
+                .into_iter()
+                .map(|(k, v)| {
+                    (
+                        k,
+                        v.into_iter()
+                            .map(|x| x.into_iter().map(|(k, v)| (k, v.into())).collect())
+                            .collect(),
+                    )
+                })
+                .collect();
+
         let registry = make_handlebars_registry();
         if let Some(template) = semver_query.per_result_error_template {
-            assert!(!actual_results.is_empty());
+            assert!(!transparent_actual_results.is_empty());
 
-            let flattened_actual_results: Vec<_> = actual_results
-                .into_iter()
+            let flattened_actual_results: Vec<_> = transparent_actual_results
+                .iter()
                 .flat_map(|(_key, value)| value)
                 .collect();
             for semver_violation_result in flattened_actual_results {
-                let pretty_result: BTreeMap<String, TransparentValue> = semver_violation_result
-                    .into_iter()
-                    .map(|(k, v)| (k, v.into()))
-                    .collect();
-
                 registry
-                    .render_template(&template, &pretty_result)
+                    .render_template(&template, semver_violation_result)
                     .with_context(|| "Error instantiating semver query template.")
                     .expect("could not materialize template");
             }
         }
+
+        if let Some(witness) = semver_query.witness {
+            let expected_witness_text =
+                std::fs::read_to_string(format!("./test_outputs/witnesses/{query_name}.output.ron"))
+                .with_context(|| format!("Query {query_name} has a witness template, but could not load\n\
+                    `test_outputs/witnesses/{query_name}.output.ron` expected witness file, did you forget to add it?"))
```

Comment:
In a separate PR, could you amend our contributing docs for adding new lints to add an optional step for "add a witness" and add a small section explaining how to do it?